framework/include/geomsearch/ElementPairInfo.h

```diff
@@ -39,16 +39,31 @@ class ElementPairInfo
 public:
 
   ElementPairInfo(const Elem * elem,
+                  const Elem * pair_elem,
```

Comment:
@jiangwen84 I think this all looks good, except that I think the naming here is a little confusing. How about `elem1`, `elem2`, `elem1_constraint_q_point`, `elem2_constraint_q_point`, etc?